microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/services/MicroProfileValidator.java

```diff
@@ -63,20 +67,32 @@ public MicroProfileValidator(MicroProfileProjectInfo projectInfo, ValuesRulesMan
 
 	public void validate(PropertiesModel document, CancelChecker cancelChecker) {
 		List<Node> nodes = document.getChildren();
+
+		// Collect names of all properties defined in the configuration file and the project information
+		Set<String> allProperties = new HashSet<>();
+		allProperties.addAll(projectInfo.getProperties().stream().map((ItemMetadata info) -> {
+			return info.getName();
+		}).collect(Collectors.toList()));
+		allProperties.addAll(nodes.stream().filter(n -> {
+			return n.getNodeType() == NodeType.PROPERTY;
+		}).map(prop -> {
+			return ((Property)prop).getPropertyNameWithProfile();
+		}).collect(Collectors.toList()));

```

Comment:
I think its best if we don't do this here. If we are in an application.properties file that has no property expressions, then it seems that we didn't need to compute this in the first place.

Maybe we can:

Check in `validatePropertyValueExpressions()` if the property is a project property by checking `MicroProfilePropertiesUtils.getProperty(propertyName, projectInfo) == null`
 like here:
https://github.com/eclipse/lsp4mp/blob/82c4798ac142efeb06499ece053407ef01009d46/microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/services/MicroProfileValidator.java#L90-L91

and if `MicroProfilePropertiesUtils.getProperty(propertyName, projectInfo) == null`, then we can check if it matches a property in the properties file.

What do you think?

Reply:
I think this is a good idea. I'll work on getting this implemented.

Reply:
I switched to using lazy loading for creating the list (only create the list right before its needed, then reuse it for future tasks). The implementation is kind of messy; I am open to suggestions on how to improve it.